src/common/IOSEnvironmentSetup.ts

```diff
@@ -91,21 +91,16 @@ export class IOSEnvironmentSetup extends BaseSetup {
     }
 
     public async isXcodeInstalled(): Promise<string> {
-        const xcodeSelectCommand: string = '/usr/bin/xcode-select -p';
+        const xcodeBuildCommand: string = 'xcodebuild -version';
```

Comment:
Use `xcodebuild` instead to determine if Xcode is installed.

If Xcode is installed then this command will output:
```
Xcode 11.7
Build version 11E801a
```


If Xcode is not installed then we'll get the following useful error message back and we'll show to the user:
```
Xcode is not installed or configured. Error: Command failed: xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance, error code: 1
```

Reply:
I think this is a concise solution to the issue. 👍

I started looking down the path of how we could provide more instruction on remediating the issue, as we've had a few people give feedback for this scenario of basically, "I didn't know what to do." The problem is, our current model of messaging isn't well set up for this.  For example, I changed `ios:reqs:xcode:unfulfilledMessage` locally to the following:

> Xcode is not installed or configured. %s\nIf Xcode is already installed, please set the CLI tools path to its location with the following steps:\n1. Open the Terminal app.\n2. Type 'sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer'\n3. Enter your password when prompted.

But the output looks like this:

> └─ Setup (0.225 sec)
   ├─ Passed (0.017 sec): @salesforce/lwc-dev-server plugin is installed.
   ├─ Passed (0.019 sec): Running macOS.
   ├─ Failed (0.015 sec): Xcode is not installed or configured. Error: Command failed: xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance, error code: 1
> If Xcode is already installed, please set the CLI tools path to its location with the following steps:
> 1. Open the Terminal app.
> 2. Type 'sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer'
> 3. Enter your password when prompted.
   └─ Passed (0.174 sec): One or more supported simulator runtimes are configured for iOS: [ 'iOS-14-1' ]
 node_modules/cli-ux/lib/styled/tree.js
ERROR running force:lightning:local:setup:  Setup for platform iOS failed. 
>
> Try this: 
Xcode is not installed or configured. Error: Command failed: xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance, error code: 1
If Xcode is already installed, please set the CLI tools path to its location with the following steps:
> 1. Open the Terminal app.
> 2. Type 'sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer'
> 3. Enter your password when prompted.
 node_modules/@salesforce/command/lib/ux.js

I dunno, maybe it's better than not having remediation steps? ¯\\_(ツ)_/¯ My ultimate takeaway is that the message that goes into the "try this" remediation section doesn't really appropriately align with `unfulfilledMessage`, in scenarios where you actually *want* to provide specific remediation steps.

I'm considering that we should add an optional `remediationMessage` property to `Requirement`, and if it's present, then it would be what's shown in the "Try this" section. (If not present, we'd go the with current behavior of overloading the message with the `unfulfilledMessage` content.) That should probably be a follow-on story (i.e. not to hold up this PR). But ultimately I think we need something like that functionality as well, for use cases like this.

Reply:
💯  on adding `remediationMessage`